tests/org.eclipse.n4js.hlc.integrationtests/src/org/eclipse/n4js/hlc/integrationtests/N4jscliTest.java

```diff
@@ -0,0 +1,163 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.hlc.integrationtests;
+
+import static org.eclipse.n4js.hlc.integrationtests.HlcTestingConstants.TARGET;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.Socket;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+
+import org.eclipse.n4js.N4JSInjectorProvider;
+import org.eclipse.n4js.binaries.BinaryCommandFactory;
+import org.eclipse.n4js.binaries.nodejs.NodeJsBinary;
+import org.eclipse.n4js.test.helper.hlc.N4CliHelper;
+import org.eclipse.n4js.utils.process.ProcessResult;
+import org.eclipse.xtext.testing.InjectWith;
+import org.eclipse.xtext.testing.XtextRunner;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.google.inject.Inject;
+
+/**
+ * The tests in this class test n4js-cli npm.
+ */
+@RunWith(XtextRunner.class)
+@InjectWith(N4JSInjectorProvider.class)
+public class N4jscliTest extends AbstractN4jscJarTest {
+
+	private static final int PORT = 4873;
+
+	private static final String LOCALHOST = "localhost";
+
+	private static final String NODE_MODULES = "node_modules";
+
+	private static final String PROJECT_NAME = "PSingleTestNpm";
+
+	@Inject
+	private NodeJsBinary nodeJsBinary;
+
+	@Inject
+	private BinaryCommandFactory commandFactory;
+
+	private final String localNpmRegstry = "http://" + LOCALHOST + ":" + PORT;
+
+	/**
+	 * Constructor
+	 *
+	 */
+	public N4jscliTest() {
+		super("fixture");
+	}
+
+	private static boolean checkHostAvailability() {
+		try (Socket s = new Socket(LOCALHOST, PORT)) {
+			return true;
+		} catch (IOException ex) {
+			// Do nothing
+		}
+		return false;
+	}
+
+	@Override
+	@Before
+	public void setupWorkspace() throws IOException {
+		// Before we make the effort to test anything, check if the npm local registry is available!
+		assertTrue("Local npm registry is not found at " + localNpmRegstry, checkHostAvailability());
+
+		super.setupWorkspace();
+		// Create a fresh node_modules folder in WSP
+		File nodeModulesFolder = new File(
+				pathToProject(PROJECT_NAME).toString() + File.separatorChar + NODE_MODULES);
+		if (nodeModulesFolder.exists()) {
+			nodeModulesFolder.delete();
+		}
+		nodeModulesFolder.mkdirs();
+
+		// Create .npmrc file in the PSingleTestNpm folder
+		File npmrcFile = new File(pathToProject(PROJECT_NAME).toString() + File.separatorChar + ".npmrc");
+		if (npmrcFile.exists()) {
+			npmrcFile.delete();
+		}
+		npmrcFile.createNewFile();
+
+		PrintWriter out = new PrintWriter(npmrcFile);
+		out.println("registry=" + localNpmRegstry);
+		out.close();
+	}
+
+	/**
+	 * Create & start a process from {@value #PROJECT_NAME}-folder.
```

Comment:
replace `&` renders weirdly in GitHub

Reply:
I think the rendering in GitHub shows that this is invalid Javadoc (like a plain `&` in html). Use `&amp;` instead.